packages/beacon-node/src/chain/historicalState/utils/stateArchive.ts

```diff
@@ -0,0 +1,106 @@
+import {ChainForkConfig} from "@lodestar/config";
+import {BeaconState, ssz} from "@lodestar/types";
+import {IStateDiffCodec} from "../types.js";
+import {getStateSlotFromBytes} from "../../../util/multifork.js";
+import {IBeaconDb} from "../../../db/interface.js";
+import {StateArchive} from "../../../db/repositories/hierarchicalStateArchive.js";
+
+export function stateToStateArchive(state: BeaconState, forkConfig: ChainForkConfig): StateArchive {
+  const slot = state.slot;
+  const {BeaconState, Balances} = forkConfig.getForkTypes(slot);
+
+  const stateRoot = BeaconState.hashTreeRoot(state);
+  const partialState = BeaconState.clone(state);
+  const balances = Balances.clone(partialState.balances);
+  partialState.balances = [];
+
+  return {
+    snapshot: true,
+    slot,
+    stateRoot,
+    partialState: BeaconState.serialize(partialState),
+    balances: Balances.serialize(balances),
+  };
+}
+
+export function stateBytesToStateArchive(stateBytes: Uint8Array, forkConfig: ChainForkConfig): StateArchive {
+  const slot = getStateSlotFromBytes(stateBytes);
+  return stateToStateArchive(forkConfig.getForkTypes(slot).BeaconState.deserialize(stateBytes), forkConfig);
```

Comment:
it's not performant to deserialize to state object here, if we have state ViewDU on main thread we should be able to compute partialState and balances there before passing through thread boundary

but if we have the full state bytes on the main thread we need to find a way to extract a full state bytes to partial state bytes and balances state bytes anyway. Maybe we can think about a ultility function to do that right on state bytes?